packages/react/src/components/Select/index.test.tsx

```diff
@@ -0,0 +1,263 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { renderToString } from 'react-dom/server'; // convert the React element to HTML
+import Select from './';
+import axe from '../../axe';
+
+const defaultProps = {
+  name: 'Test Select',
+  label: 'Test Select'
+};
+
+const withCustomOptions = (otherProps = {}) => {
+  const options = [
+    { key: '1', value: 'Fred' },
+    { key: '2', value: 'Bill' },
+    { key: '3', value: 'Ted' },
+    { key: '4', value: 'Bob' }
+  ];
+
+  return render(
+    <div>
+      <Select
+        {...defaultProps}
+        {...otherProps}
+        defaultValue="Bill"
+        options={options}
+      />
+    </div>
+  );
+};
+
+test('Should render the expected UI elements', () => {
+  withCustomOptions();
+
+  expect(screen.getByText(defaultProps.name)).toBeInTheDocument();
+  expect(
+    screen.getByText(defaultProps.name).parentElement?.parentElement
+  ).toHaveClass('Field__select');
+  expect(screen.getAllByRole('option')[0]).toHaveClass('Field__option');
+  expect(screen.getAllByRole('option')).toHaveLength(4);
+  expect(screen.queryByText('Required')).toBeNull();
+  expect(screen.queryByText('Error')).toBeNull();
+});
+
+test('Should set option attributes properly', () => {
+  render(
+    <Select
+      {...defaultProps}
+      defaultValue="a"
+      options={[
+        { key: '1', value: 'a' },
+        { disabled: true, key: '2', value: 'b' },
+        { key: '3', value: 'c' }
+      ]}
+    />
+  );
+
+  expect(screen.getAllByRole('option')).toHaveLength(3);
+  expect(screen.getByText('b')).toBeInTheDocument();
```

Comment:
This checks that it exists in the document, but we probably want something more strict here such as there's an option role with the desired text.